File: academic-system/src/main/java/org/example/service/ServicoRelatorio.java
package org.example.service;

import java.util.List;
import org.example.model.Turma;
import org.example.model.Avaliacao;

public class ServicoRelatorio {
    
    public String gerarResumoAvaliacoes(List<Turma> turmas, String usuarioLogado) {
        StringBuilder sb = new StringBuilder();
        sb.append("--- Relatório de Resumo de Avaliações ---\n");
        
        if (turmas == null || turmas.isEmpty()) {
            sb.append("Nenhuma turma registrada.\n");
            return sb.toString();
        }

        for (Turma t : turmas) {
            sb.append("Turma: ").append(t.getCodigo()).append(" - ").append(t.getTitulo()).append("\n");
            
            List<Avaliacao> avaliacoes = t.getAvaliacoes();
            if (avaliacoes.isEmpty()) {
                sb.append("  Sem avaliações registradas.\n");
            } else {
                for (Avaliacao a : avaliacoes) {
                    sb.append("  Avaliação: ")
                      .append(a.getClass().getSimpleName())
                      .append(" | Valor: ").append(a.getValor())
                      .append(" | Peso: ").append(a.getPeso())
                      .append("\n");
                }
            }
        }
        return sb.toString();
    }

    public String gerarRelatorioPesos(List<Turma> turmas, String usuarioLogado) {
        StringBuilder sb = new StringBuilder();
        sb.append("--- Relatório de Pesos das Avaliações ---\n");
        
        if (turmas == null || turmas.isEmpty()) {
            sb.append("Nenhuma turma registrada.\n");
            return sb.toString();
        }

        for (Turma t : turmas) {
            double pesoTotal = 0.0;
            for (Avaliacao a : t.getAvaliacoes()) {
                pesoTotal += a.getPeso();
            }
            
            sb.append("Turma: ").append(t.getCodigo()).append(" - ").append(t.getTitulo()).append("\n");
            sb.append("  Peso Total: ").append(String.format(java.util.Locale.US, "%.1f", pesoTotal)).append("\n");
            
            if (Math.abs(pesoTotal - 1.0) < 0.0001) {
                sb.append("  Status: Composição VÁLIDA\n");
            } else {
                sb.append("  Status: Composição INVÁLIDA\n");
            }
        }
        return sb.toString();
    }

    public String gerarRelatorioPersistencia(String tipoAtual, String usuarioLogado) {
        if (!"ADMIN".equals(usuarioLogado)) {
            throw new org.example.exception.ExcecaoAutorizacao("Operação negada: Apenas administradores podem gerar relatório de persistência.");
        }
        return "--- Relatório de Configuração de Persistência ---\nTipo de Persistência Atual: " + tipoAtual + "\n";
    }
}

File: academic-system-csharp/src/SistemaAcademico/Services/ServicoRelatorio.cs
using System.Globalization;
using System.Text;
using SistemaAcademico.Models;
using SistemaAcademico.Exceptions;
using Microsoft.Extensions.Logging;

namespace SistemaAcademico.Services;

// Equivalente à classe Java: ServicoRelatorio
// java.util.logging.Logger → Microsoft.Extensions.Logging.ILogger
// String.format(Locale.US, "%.1f", pesoTotal) → pesoTotal.ToString("F1", CultureInfo.InvariantCulture)
public class ServicoRelatorio
{
    private static readonly ILogger<ServicoRelatorio> logger =
        LoggerFactory.Create(b => b.AddConsole()).CreateLogger<ServicoRelatorio>();

    public string GerarResumoAvaliacoes(List<Turma> turmas, string usuarioLogado)
    {
        logger.LogInformation("Relatório de Resumo de Avaliações gerado pelo usuário: {Usuario}", usuarioLogado);
        StringBuilder sb = new StringBuilder();
        sb.Append("--- Relatório de Resumo de Avaliações ---\n");

        if (turmas == null || turmas.Count == 0)
        {
            sb.Append("Nenhuma turma registrada.\n");
            return sb.ToString();
        }

        foreach (Turma t in turmas)
        {
            sb.Append("Turma: ").Append(t.Codigo).Append(" - ").Append(t.Titulo).Append("\n");

            List<Avaliacao> avaliacoes = t.GetAvaliacoes();
            if (avaliacoes.Count == 0)
            {
                sb.Append("  Sem avaliações registradas.\n");
            }
            else
            {
                foreach (Avaliacao a in avaliacoes)
                {
                    sb.Append("  Avaliação: ")
                      .Append(a.GetType().Name)
                      .Append(" | Valor: ").Append(a.Valor)
                      .Append(" | Peso: ").Append(a.Peso)
                      .Append("\n");
                }
            }
        }
        return sb.ToString();
    }

    public string GerarRelatorioPesos(List<Turma> turmas, string usuarioLogado)
    {
        logger.LogInformation("Relatório de Pesos das Avaliações gerado pelo usuário: {Usuario}", usuarioLogado);
        StringBuilder sb = new StringBuilder();
        sb.Append("--- Relatório de Pesos das Avaliações ---\n");

        if (turmas == null || turmas.Count == 0)
        {
            sb.Append("Nenhuma turma registrada.\n");
            return sb.ToString();
        }

        foreach (Turma t in turmas)
        {
            double pesoTotal = 0.0;
            foreach (Avaliacao a in t.GetAvaliacoes())
            {
                pesoTotal += a.Peso;
            }

            sb.Append("Turma: ").Append(t.Codigo).Append(" - ").Append(t.Titulo).Append("\n");
            // Equivalente ao String.format(Locale.US, "%.1f", pesoTotal)
            sb.Append("  Peso Total: ").Append(pesoTotal.ToString("F1", CultureInfo.InvariantCulture)).Append("\n");

            if (Math.Abs(pesoTotal - 1.0) < 0.0001)
            {
                sb.Append("  Status: Composição VÁLIDA\n");
            }
            else
            {
                sb.Append("  Status: Composição INVÁLIDA\n");
            }
        }
        return sb.ToString();
    }

    public string GerarRelatorioPersistencia(string tipoAtual, string usuarioLogado)
    {
        if (!"ADMIN".Equals(usuarioLogado))
        {
            logger.LogWarning("Falha de autorização: Tentativa de gerar relatório de persistência por {Usuario}", usuarioLogado);
            throw new ExcecaoAutorizacao("Operação negada: Apenas administradores podem gerar relatório de persistência.");
        }
        logger.LogInformation("Relatório de Persistência gerado pelo administrador.");
        return "--- Relatório de Configuração de Persistência ---\nTipo de Persistência Atual: " + tipoAtual + "\n";
    }
}
